### PR TITLE
[ARTEMIS-444] Extend AIOSequentialFileFactory to check underlying FS

### PR DESCRIPTION
**Pull request by dpospisil**
https://issues.apache.org/jira/browse/ARTEMIS-444

### CODE REVIEW

**Comment by clebertsuconic** [comment]
the method seems right, but you're never using the new method you added?

**Comment by clebertsuconic** [comment]
ActiveMQServer::initialisePart1 needs this change: 

   synchronized boolean initialisePart1(boolean scalingDown) throws Exception {
      if (state == SERVER_STATE.STOPPED)
         return false;

      // Create the pools - we have two pools - one for non scheduled - and another for scheduled
      initializeExecutorServices();

      if (configuration.getJournalType() == JournalType.ASYNCIO && !AIOSequentialFileFactory.isSupported(configuration.getJournalDirectory())) {

**Comment by andytaylor** [comment]
Can you improve this so the log message makes sense and to actually use the method. so something like 'LIBAIO is installed but /path/somewhere does not support AIO'.

**Comment by clebertsuconic** [comment]
+1 on adding a message with anything saying the device on path ('/actualpath') doesn't support libaio, switching to NIO.

**Comment by dpospisil** [comment]
I have updated the PR so the ActiveMQServer uses it as suggested.

I have also improved the check so that it returns false (unsupported) only if AIO check fails and plain java pass.

File: artemis-journal/src/main/java/org/apache/activemq/artemis/core/io/aio/AIOSequentialFileFactory.java [hunk 1]
```diff
@@ -114,6 +116,36 @@ public static boolean isSupported() {
...
+         LibaioContext.close(fd);
+         aioTestFile.delete();
+      }
+      catch (Exception e) {
+         // try to handle the file using plain Java
+         // return false if and only if we can create/remove the file using
+         // plain Java but not using AIO
+         try {
+            if (!aioTestFile.exists()) {
+               if (!aioTestFile.createNewFile()) return true;
```
Comment (clebertsuconic) [inline artemis-journal/src/main/java/org/apache/activemq/artemis/core/io/aio/AIOSequentialFileFactory.java]: Isn't that a bit convoluted?

**Comment by dpospisil** [comment]
I have tried to avod NIO swich as much as possible. My concern was a situation where e.g. file privilegues would block access to the journal which would make AIO open to fail and than fallback to NIO. But If you don't like it let me know I'll remove it.

**Pull request closed without merging**